py-polars/polars/utils/_construction.py

```diff
@@ -875,9 +875,9 @@ def _expand_dict_scalars(
                 elif val is None or isinstance(  # type: ignore[redundant-expr]
                     val, (int, float, str, bool, date, datetime, time, timedelta)
                 ):
-                    updated_data[name] = pl.Series(
-                        name=name, values=[val], dtype=dtype
-                    ).extend_constant(val, array_len - 1)
+                    updated_data[name] = F.repeat(
+                        val, array_len, dtype=dtype, eager=True
+                    ).alias(name)
```

Comment:
This is a drive-by change I ran into. `repeat` is a bit more efficient because it ends up with 1 chunk rather than 2.